rdagent/scenarios/data_science/proposal/exp_gen/merge.py

```diff
@@ -304,21 +304,37 @@ def __init__(self, *args, **kwargs):
         )
         self.flag_start_merge = False
 
-    def gen(self, trace: DSTrace) -> DSExperiment:
+    def reset_exp_gen_version(self, version: str = "v2"):
+        DS_RD_SETTING.proposal_version = version
```

Comment:
Changing the setting only will not switch the proposal instance now.
Please check it